apps/web-giddh/src/app/vat-report/transactions/vatReportTransactions.component.ts

```diff
@@ -0,0 +1,300 @@
+    }
+
+    /**
+     * This will get called on close model event
+     *
+     * @param {{ action: string }} userResponse
+     * @memberof VatReportTransactionsComponent
+     */
+    public closeDownloadOrSendMailPopup(userResponse: { action: string }): void {
+        this.downloadOrSendMailModel.hide();
+        if (userResponse.action === 'update') {
+            this.store.dispatch(this.invoiceActions.VisitToInvoiceFromPreview());
+            this.invoiceGenerateModel.show();
+        } else if (userResponse.action === 'closed') {
+            this.store.dispatch(this.invoiceActions.ResetInvoiceData());
+        }
+    }
+
+    /**
+     * This will get call on close of invoice popup
+     *
+     * @param {*} e
+     * @memberof VatReportTransactionsComponent
+     */
+    public closeInvoiceModel(e): void {
+        this.invoiceGenerateModel.hide();
+        setTimeout(() => {
+            this.store.dispatch(this.invoiceActions.ResetInvoiceData());
+        }, 2000);
+    }
+
+    /**
+     * This will get called on download/send email
+     *
+     * @param {{ action: string, emails: string[], numbers: string[], typeOfInvoice: string[] }} userResponse
+     * @memberof VatReportTransactionsComponent
+     */
+    public onDownloadOrSendMailEvent(userResponse: { action: string, emails: string[], numbers: string[], typeOfInvoice: string[] }): void {
+        if (userResponse.action === 'download') {
+            this.downloadFile();
+        } else if (userResponse.action === 'send_mail' && userResponse.emails && userResponse.emails.length) {
+            this.store.dispatch(this.invoiceActions.SendInvoiceOnMail(this.selectedInvoice.accountUniqueName, {
+                emailId: userResponse.emails,
+                voucherNumber: [this.selectedInvoice.voucherNumber],
+                typeOfInvoice: userResponse.typeOfInvoice,
+                voucherType: this.selectedInvoice.voucherType
+            }));
+        } else if (userResponse.action === 'send_sms' && userResponse.numbers && userResponse.numbers.length) {
+            this.store.dispatch(this.invoiceActions.SendInvoiceOnSms(this.selectedInvoice.account.uniqueName, { numbers: userResponse.numbers }, this.selectedInvoice.voucherNumber));
+        }
+    }
+
+    /**
+     * This will get called on download invoice
+     *
+     * @param {*} invoiceCopy
+     * @memberof VatReportTransactionsComponent
+     */
+    public ondownloadInvoiceEvent(invoiceCopy): void {
+        let dataToSend = {
+            voucherNumber: [this.selectedInvoice.voucherNumber],
+            typeOfInvoice: invoiceCopy,
+            voucherType: this.selectedInvoice.voucherType
+        };
+
+        this.invoiceService.DownloadInvoice(this.selectedInvoice.accountUniqueName, dataToSend)
+            .subscribe(res => {
+                if (res) {
+                    if (dataToSend.typeOfInvoice.length > 1) {
```

Comment:
Check here does we need .zip and .PDF both in case of multiple invoice selected?